classes.ipynb

```diff
@@ -0,0 +1,1597 @@
+       "\n",
+       "[4878 rows x 40 columns]"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Download the historical stock data for each ETF and store it in a dictionary\n",
+    "etf_dict = {}\n",
+    "for etf in ETFList:\n",
+    "    etf_dict[etf] = yf.download(etf)\n",
+    "\n",
+    "# Create a new DataFrame to store the closing prices of each ETF\n",
+    "etf_close_data = pd.DataFrame()\n",
+    "\n",
+    "# Extract the 'Close' column for each ETF and assign it as a new column in etf_data\n",
+    "for etf in ETFList:\n",
+    "    etf_close_data[etf] = etf_dict[etf]['Close']\n",
+    "\n",
+    "# Print the resulting DataFrame\n",
+    "etf_close_data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['VGT', 'IVV', 'VTI']"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "stockChoices"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Class #2\n",
+    "\n",
+    "class trade:\n",
+    "    \"\"\"\n",
+    "    This class will preform most of the trading functionality. Will be added onto when another trading function is added.\n",
+    "\n",
+    "    Attributes:\n",
+    "\n",
+    "\n",
+    "    \"\"\"\n",
+    "\n",
+    "    def __init__(self) -> None:\n",
+    "        self.investment_amount = 1000 #init a random value, functions take care of actual value init.\n",
+    "        self.frequency = input('How often do you want the trades to occur (daily, weekly, biweekly, monthly)?')\n",
+    "        self.investment_day = input('Which day of the week do you want the trading to occur?')\n",
+    "        self.stock_exp_ratio = 0 #init as 0, and will update when the function is run.\n",
+    "        self.start_date = '2017-01-01'\n",
+    "        self.end_date = '2022-12-31'\n",
+    "        self.stock_data = stockData\n",
+    "        self.etf_dict = {}\n",
+    "\n",
+    "    def historicalData(self):\n",
+    "        \n",
+    "        for etf in ETFList:\n",
+    "            self.etf_dict[etf] = yf.download(etf)\n",
+    "\n",
+    "        # Create a new DataFrame to store the closing prices of each ETF\n",
+    "        etf_close_data = pd.DataFrame()\n",
+    "\n",
+    "        # Extract the 'Close' column for each ETF and assign it as a new column in etf_data\n",
+    "        for etf in ETFList:\n",
+    "            etf_close_data[etf] = self.etf_dict[etf]['Close']\n",
+    "\n",
+    "        return etf_close_data\n",
+    "\n",
+    "        \n",
+    "\n",
+    "    def dollar_cost_average(stock_data, investment_amount, frequency, investment_day, stock_exp_ratio, start_date, end_date):\n",
```

Comment:
Please add the data type in the function declaration, such as stock_data: pd.DataFrame